File: frontend/src/app/models/match/create-match-event.request.ts
export type MatchEventType = 'GOAL' | 'YELLOW_CARD' | 'RED_CARD' | 'SUBSTITUTION' | 'OTHER';

export interface CreateMatchEventRequest {
  playerId?: number | null;
  secondaryPlayerId?: number | null;
  teamId: number;
  eventType: MatchEventType;
  minute: number;
  description?: string | null;
}

File: src/main/java/com/tournament/app/footycup/backend/dto/match/CreateMatchEventRequest.java
package com.tournament.app.footycup.backend.dto.match;

import jakarta.validation.constraints.NotNull;

public record CreateMatchEventRequest(
        Long playerId,
        Long secondaryPlayerId,
        @NotNull Long teamId,
        @NotNull String eventType,
        @NotNull Integer minute,
        String description
) {
}
